Docs adding uninstall instructions

kostickm opened the pull request:
Added uninstall instructions for Docker and Boot2Docker where applicable. Fixes #12824 

This included the following operating systems: Ubuntu, Mac OS X, Windows, Arch Linux, CentOS, CRUX Linux, Debian, Fedora, FrugalWare, Gentoo, RHEL, Oracle, SUSE

I don't have access to all OSs, so feedback is greatly appreciated!

Signed-off-by: Megan Kostick mkostick@us.ibm.com

kostickm committed:
Docs adding uninstall instructions

Signed-off-by: Megan Kostick <mkostick@us.ibm.com>

thaJeztah commented on docs/sources/installation/archlinux.md:
```diff
@@ -59,3 +59,13 @@ To start on system boot:
+
+To uninstall the Docker package:
+
+	pacman -R docker
```
Missing `$` prompt. Does this require `sudo`?

thaJeztah commented on docs/sources/installation/archlinux.md:
```diff
@@ -59,3 +59,13 @@ To start on system boot:
+
+To uninstall the Docker package and dependencies that are no longer needed:
+
+	pacman -Rns docker
```
Same here.

Also (not sure) indentation looks to be > 4 spaces (here and line 67)

thaJeztah commented on docs/sources/installation/cruxlinux.md:
```diff
@@ -60,6 +60,13 @@ or use it as part of your `FROM` line in your `Dockerfile(s)`.
+
+To uninstall the Docker package:
+
+    # prt-get remove docker
```
`#` prompt indicates that this is run as `root`. To be consistent, I think it should be `$ sudo ..`

thaJeztah commented on docs/sources/installation/fedora.md:
```diff
@@ -42,6 +48,12 @@ To update the `docker` package:
 
+### Uninstall Docker
+
+To uninstall the `docker` package:
```
The `docker` package is in back-ticks here, which (I think) hints at it being the literal name of the software that's removed. However, the package here is called `docker-io`, so this might become confusing.

Perhaps we should just use "Docker package" in all documents, to be consistent and not refer to the  actual name of the package.

/cc @docker/docs-owners

thaJeztah replied:
Hm, not sure that worked;

/cc @moxiegirl @fredlf @SvenDowideit @jamtur01

fredlf replied:
Agreed, I prefer "Docker package" unless we are literally referring to a typed command or file/executable.

moxiegirl replied:
I agree with @thaJeztah just Docker package works.

thaJeztah commented on docs/sources/installation/frugalware.md:
```diff
@@ -48,3 +48,13 @@ To start on system boot:
+
+To uninstall the Docker package and dependencies that are no longer needed:
+
+    pacman -Rns lxc-docker
```
Same here; missing `$` (and `sudo`?)

thaJeztah commented on docs/sources/installation/mac.md:
```diff
@@ -314,6 +314,31 @@ section.
+
+7. Run the uninstall.sh script:
+
+		$ ./uninstall.sh
```
Indentation here (and above) looks > 4 spaces (perhaps you're using a tab here?)

I'd prefer to use spaces, because that's used in (most) places

thaJeztah replied:
Note; perhaps 8 spaces is needed because of the numbered list here (in that case, ignore my comment)

thaJeztah commented on docs/sources/installation/oracle.md:
```diff
@@ -99,6 +99,12 @@ To enable btrfs support on Oracle Linux:
+
+To uninstall the Docker package:
+
+	$ sudo yum -y remove docker
```
> 4 spaces or tab?

thaJeztah commented on docs/sources/installation/ubuntulinux.md:
```diff
@@ -310,3 +310,13 @@ To install the latest version of Docker, use the standard `-N` flag with `wget`:
+
+To uninstall the Docker package and dependencies that are no longer needed:
+
+	$ sudo apt-get autoremove lxc-docker
```
> 4 spaces or tab? (Also on line 317)

thaJeztah commented on docs/sources/installation/windows.md:
```diff
@@ -150,6 +150,12 @@ You can do this with
+
+Boot2Docker can be uninstalled using the normal Windows process for uninstalling
+a program. This will not delete the `docker-install.exe`, but you can simply 
+delete the file at any time.
```
s/the/that/

moxiegirl replied:
You can uninstall Boot2Docker using Window's standard process for removing programs.  This process does not remove the `docker-install.exe` file. You must delete that file yourself.

(The any time is kind of assumed and a single sentence is stronger using the reflexive pronoun.)

thaJeztah commented:
Thanks @kostickm! Added some notes from a docs perspective, but I haven't been able to test the actual procedures yet, so I hope some of the maintainers / contributors is able to test them.

RickWieman commented:
For e.g. `apt-get` there's a difference between `remove docker` and `purge docker`... I'm not sure which of the two should be used in this uninstall manual (i.e. I don't know if we just want to help the user to simply uninstall Docker (and dependencies) or to also help to remove the configuration files etc.).

thaJeztah commented:
> remove docker and purge docker... I'm not sure which of the two should be used in this uninstall manual

Good point. I think that (if https://github.com/docker/docker/issues/12418#issuecomment-95993896 is indeed true) `purge` would be a safe option.

Thinking of that, the uninstall instructions should probably mention images, containers and volumes;
- Are they removed during the uninstall?
- If *not*, how to do that (and/or where to find them)
- If they *are* removed, warn the reader that they will lose that data.

kostickm commented:
@thaJeztah Thank you for the comments! On some of the files I kept it consistent with what was currently in the file though it didn't quite follow the doc styling rules I read. But I will go through and make the updates.

As far as the remove vs. purge, I wasn't sure there either what we wanted to say. I will add some more details there to include additional info for the user.

fredlf commented on docs/sources/installation/SUSE.md:
```diff
@@ -28,7 +28,7 @@ Docker is available in **SUSE Linux Enterprise 12 and later**. Please note that
 architecture.
 
-# Installation
+## Installation
```
Ideally, we should be consistent with tense, etc. in headings. "Install Docker" "Uninstall Docker" or "Installation" "Uninstallation", etc.

kostickm commented:
@fredlf I agree, I will change them to all be consistent!

moxiegirl commented on docs/sources/installation/rhel.md:
```diff
@@ -119,6 +119,20 @@ Docker runtime files, or make other customizations, read our Systemd article to
 learn how to [customize your Systemd Docker daemon options](/articles/systemd/).
 
 
+## Red Hat Enterprise Linux 7 Uninstall Docker
```
Minor thing (very) but as you have @thaJeztah's comments already, `Uninstall` should be lowercase.   Why? Our style guidance is sentence case --- proper nouns still capped of course.

kostickm commented:
@moxiegirl Oops you are right, I missed that one. Will update!

moxiegirl commented:
@kostickm Really nice, well-though contribution. Thanks!

kostickm commented:
I've gone through and made the updates you all suggested. Thank you all for the feedback! There were quite a few files using tabs instead of spaces, so I have updated those areas as well.

I added a note about the images, containers, etc, but am not sure if this is how we want to word it. So will need some input here.

thaJeztah commented:
Thanks! I don't have time to look through the changes right now; for those that want to see the textual changes and *not* the whitespace changes, use this link: https://github.com/docker/docker/pull/12882/files?w=1

moxiegirl commented on docs/sources/installation/SUSE.md:
```diff
@@ -76,6 +76,20 @@ If you need to add an HTTP Proxy, set a different directory or partition for the
+
+    $ sudo zypper rm docker
+
+> **Note**: The above command will not remove images, containers, volumes, or user
```
You shouldn't need a note for this. Basically notes:
![image](https://cloud.githubusercontent.com/assets/1347209/7424045/7708bf1c-ef51-11e4-8477-797728a376c5.png)

Since the topic is uninstalling, this applies directly.  So, I'd just:

https://gist.github.com/moxiegirl/b90a8759ebd115f08026

SvenDowideit commented:
very very awesome. LGTM!

moxiegirl commented:
@kostickm Megan, you've done a great job. I'm inclined to just carry the work to rejig the note.  Would you like to fix the note or are you cool with a carry?

kostickm commented:
@moxiegirl I was just in the middle of fixing it, so I don't mind finishing it up.

moxiegirl commented:
Super. Thank you.

kostickm commented:
@moxiegirl No problem! Re-factored the notes to plain text, so should be good to go :)

moxiegirl commented:
LGTM! Thanks again.

moxiegirl commented:
Crap! Too much to do today Megan.  I pulled the trigger prematurely.  Very sorry.  If you want to finish your last commit, I'll pull it into a branch and send it through.

kostickm commented:
@moxiegirl I think you got it. The commit you merged has the removal of the "Note" sections to regular text. Unless there is something else you think needed changed?

moxiegirl commented:
@kostickm Yep, I'm cool. I was doing a complex merge in the foreground and watching for your PR to clear the builds in the background. It was a bit of a cluster, too many browser windows and not enough coffee.  :-D 

You were very patient thank you.

kostickm commented:
@moxiegirl I know that not enough coffee feeling :) Glad it's all sorted out. Thank you for all your help on this one!

thaJeztah commented:
Thanks Megan!


> On 1 mei 2015, at 12:41, Megan Kostick <notifications@github.com> wrote:
> 
> @moxiegirl I know that not enough coffee feeling :) Glad it's all sorted out. Thank you for all your help on this one!
> 
> —
> Reply to this email directly or view it on GitHub.
>
